### ISSUE TITLE
DietPi Software | Amiberry next version

### ISSUE THREAD

**Issue by midwan** [MichaIng/DietPi#1410]
## Creating a bug report/issue:
Prepare DietPi for the upcoming release of Amiberry

#### Required Information:
- SBC Device: RPI1/2/3, XU4, OrangePi (maybe more can run it, but those are confirmed for now)

#### Additional Information (if applicable):
- Extra lib dependencies to support CD32 FMV
- SDL2, SDL2_image, SDL2_ttf (compiled from source to support KMS!)
- SDL1 version is also supported, both SDL1 and SDL2 build from the same Makefile
- new file dependencies included in the data/ folder
- No need to launch from under X11 anymore

#### Extra details:
Full list of required packages to build from source:
`libxml2-dev libflac-dev libmpg123-dev libpng-dev libmpeg2-4-dev`

Additionally, for the SDL1 version:
`libsdl-dev libguichan-dev libsdl-ttf2.0-dev libsdl-gfx1.2-dev`

Or for the SDL2 version:
- SDL2-2.0.x compiled from source to support the KMSDRM driver (or RPI driver)
- SDL2_ttf
- SDL2_image

I wrote a wiki on how to configure/compile SDL2 from source here: https://github.com/midwan/amiberry/wiki/Compile-SDL2-from-source

**Comment by Fourdee** [comment]
@midwan 

Excellent, you've been very busy with Github and great to see this project evolve. 
Many thanks for the info. I'll take a look at this over the next few days. Should be able to get it in for DietPi v6.0 release if all goes to plan 😄 

ref: https://github.com/Fourdee/DietPi/issues/1385

**Comment by Fourdee** [comment]
Notes:
- ARMv8 (Odroid C2) Jessie:
```
root@DietPi:~#  apt-get install gir1.2-ibus-1.0 libdbus-1-dev libegl1-mesa-dev libibus-1.0-5 libibus-1.0-dev libice-dev libsm-dev libsndio-dev libwayland-bin libwayland-dev libxi-dev libxinerama-dev libxkbcommon-dev libxrandr-dev libxss-dev libxt-dev libxv-dev x11proto-randr-dev x11proto-scrnsaver-dev x11proto-video-dev x11proto-xinerama-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package libwayland-bin
```

```
wget http://ftp.us.debian.org/debian/pool/main/w/wayland/libwayland-bin_1.12.0-1_arm64.deb
dpkg -i *.deb
rm *.deb
```

```
root@DietPi:~# dpkg -i *.deb
Selecting previously unselected package libwayland-bin.
dpkg: considering removing libwayland-dev in favour of libwayland-bin ...
dpkg: no, cannot proceed with removal of libwayland-dev (--auto-deconfigure will help):
 libegl1-mesa-dev:arm64 depends on libwayland-dev (>= 1.2.0)
  libwayland-dev is to be removed.

dpkg: regarding libwayland-bin_1.12.0-1_arm64.deb containing libwayland-bin:
 libwayland-bin conflicts with libwayland-dev (<< 1.11.0-1)
  libwayland-dev (version 1.6.0-2) is present and installed.

dpkg: error processing archive libwayland-bin_1.12.0-1_arm64.deb (--install):
 conflicting packages - not installing libwayland-bin
Errors were encountered while processing:
 libwayland-bin_1.12.0-1_arm64.deb
```

Although should be fine with just ```-dev``` package, will test

🈯️ removal of ```libwayland-bin```
```
apt-get install gir1.2-ibus-1.0 libdbus-1-dev libegl1-mesa-dev libibus-1.0-5 libibus-1.0-dev libice-dev libsm-dev libsndio-dev libwayland-dev libxi-dev libxinerama-dev libxkbcommon-dev libxrandr-dev libxss-dev libxt-dev libxv-dev x11proto-randr-dev x11proto-scrnsaver-dev x11proto-video-dev x11proto-xinerama-dev
```

**Comment by midwan** [comment]
We don't really use wayland anyway, so this shouldn't be a problem. The list is what I got if I tried to install the package from the official repo (which you normally don't want, because their binaries are older and don't have the RPI or KMSDRM drivers enabled)

**Comment by Fourdee** [comment]
@midwan 

```./configure --disable-video-rpi --disable-video-x11 --enable-video-kmsdrm; make -j $(nproc --all); sudo make install```

```make -j $(nproc --all)``` compiles using all available CPU cores, much quicker 😃

**Comment by midwan** [comment]
yeah, I usually specify the number of cores manually  here, but I noticed that the Pi can easily overheat if using all 4, and then it throttles down (so it becomes sloweeeeeer). I found that using up to 3 is ok.

**Comment by Fourdee** [comment]
@midwan 

> yeah, I usually specify the number of cores manually here, but I noticed that the Pi can easily overheat if using all 4, and then it throttles down (so it becomes sloweeeeeer). I found that using up to 3 is ok.

Good point, she does run a little hot lol 👍 

Want me to revert the changes: https://github.com/midwan/amiberry/wiki/Compile-SDL2-from-source ?

**Comment by midwan** [comment]
No, it's ok I think... SDL2 is not extremely large (like some Amiberry's files), so it might be able to make it before hitting the threshold. :)

**Comment by Fourdee** [comment]
> ...and SDL2_ttf:

```
configure: error:
*** Unable to find FreeType2 library (http://www.freetype.org/)
```

🈯️ ```libfreetype6-dev``` pre-req
```
apt-get install libfreetype6-dev -y
```

---

Fails on make
```
/usr/bin/ld: cannot find -lGL
collect2: error: ld returned 1 exit status
Makefile:474: recipe for target 'glfont' failed
make: *** [glfont] Error 1
make: *** Waiting for unfinished jobs....
```

🈯️ ```libgl1-mesa-dev``` pre-req
```
apt-get install libgl1-mesa-dev -y
```

---

ARMv7 ARMv8 SDL2 compiles/```.debs``` done, will continue rest tomorrow.

PATH issue?
```
src/sdl/sdlimageloader.cpp:63:23: fatal error: SDL_image.h: No such file or directory

root@DietPi:~/amiberry-dev# find / -name 'SDL_image.h'
/root/SDL2_image-2.0.2/SDL_image.h
/usr/include/SDL/SDL_image.h
/usr/include/SDL2/SDL_image.h
```

**Comment by Fourdee** [comment]
@midwan 

Apologies, will need to nudge this to v6.1 release. We have too many outstanding items for v6.0 release on Sunday.

In the mean time, i'll create a new AmiBerry image, which will use the existing installation. I'll also update you with the details once its completed, as a few things have changed in DietPi (eg: automation options).

**Comment by midwan** [comment]
@Fourdee 
No worries, we're also busy with a few things :)

**Comment by midwan** [comment]
@Fourdee 
We also have a new target: ASUS Tinker board.
However, it's still a little tricky to get it working there, as you probably noticed also. Due to the lack of proper support, we ended up compiling our own kernel (based on Armbian), using the GPU blobs and several other modifications to get SDL2 to compile with support for OpenGL ES2.

The Makefile includes the target, and the code has the necessary logic to accommodate for it (e.g. different timings, custom cursor workaround for Mali GPUs, etc). We can also use multi-threaded rendering in SDL2, which improves the performance a lot.

Sadly, multi-threaded rendering does not work on the current implementation of the KMSDRM backend for the Pi. It does work if you're running under x11 only (and you've compiled SDL2 without disabling x11 of course). So the fastest target for the Pi is still using Dispmanx directly.

**Comment by Fourdee** [comment]
@midwan 

Thanks for the info 👍  

I'll try and pick this back up over next few days, got side tracked with bug fixes etc.

**Comment by midwan** [comment]
@Fourdee 
I've asked the person who worked on the port to get in touch with you about using the custom kernel and extra modifications used on the Tinkerboard, since we're using DietPi as a basis now :)

I think getting GPU support on the Tinkerboard using DietPi would be beneficial to all...

**Comment by Fourdee** [comment]
@midwan 
> I think getting GPU support on the Tinkerboard using DietPi would be beneficial to all...

Agree.

We'll focus on RPi 1st, then Odroid and ASUSTB.

**Comment by midwan** [comment]
@Fourdee 
BTW, here's the repo with the kernel used, config and info about the extra stuff needed (gpu blobs etc):
https://github.com/alynna/tinker-kdev

It has been tested to work on DietPi. ;-)

**Comment by Fourdee** [comment]
@midwan 

#### Testing previous and dev AmiBerry binaries on RPi3 Stretch, with SDL2 installed as per your guide.
```
INFO: SDL could not initialize! SDL_Error: No Available video device
```
Both with/without ```xinit```

#### Under RPi OpenGL, using compiled SDL2 as per your guide:
- AmiBerry-sdl2 (old) is fine under ```xinit```
- AmiBerry-sdl2 (dev) under xinit, ```MESA-LOADER: failed to retrieve device info```.

---

@midwan 
~```./configure --disable-video-rpi --enable-video-kmsdrm```, should this not be ```./configure  --disable-video-opengl``` only for RPi, if we are targeting OpenGLES?~

Tried with ```./configure --host=arm-raspberry-linux-gnueabihf --disable-video-opengl #--disable-video-rpi --enable-video-kmsdrm```
Same results

---

### Here is what I have so far, SDL2 only:
NB: DietPi system, else replace
```G_AGI``` = ```apt-get install```
```G_RUN_CMD``` = remove this
```
#AmiBerry compile  -----------------------------------------------------------------

#SDL2
G_AGI build-essential libfreetype6-dev gir1.2-ibus-1.0 libdbus-1-dev libegl1-mesa-dev libgl1-mesa-dev libibus-1.0-5 libibus-1.0-dev libice-dev libsm-dev libsndio-dev libwayland-dev libxi-dev libxinerama-dev libxkbcommon-dev libxrandr-dev libxss-dev libxt-dev libxv-dev x11proto-randr-dev x11proto-scrnsaver-dev x11proto-video-dev x11proto-xinerama-dev
#libwayland-bin

cd "$HOME"
wget https://www.libsdl.org/release/SDL2-2.0.7.tar.gz
tar zxvf SDL2-2.0.7.tar.gz
cd SDL2-2.0.7
./configure --disable-video-rpi --enable-video-kmsdrm #--host=arm-raspberry-linux-gnueabihf #--disable-video-opengl
G_RUN_CMD make -j $(nproc --all)
make install

cd "$HOME"
wget https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.2.tar.gz
tar zxvf SDL2_image-2.0.2.tar.gz
cd SDL2_image-2.0.2
mkdir build
cd build
../configure
G_RUN_CMD make -j $(nproc --all)
make install

cd "$HOME"
wget https://www.libsdl.org/projects/SDL_ttf/release/SDL2_ttf-2.0.14.tar.gz
tar zxvf SDL2_ttf-2.0.14.tar.gz
cd SDL2_ttf-2.0.14
mkdir build
cd build
../configure
G_RUN_CMD make -j $(nproc --all)
make install

#	LDconfig for /usr/local/lib, else, fails to load SDL2-ttf.so binary
echo -e "/usr/local/lib" > /etc/ld.so.conf.d/usr-local-lib.conf
ldconfig

#AmiBerry

G_AGI build-essential git \
libxml2-dev libflac-dev libmpg123-dev libpng-dev libmpeg2-4-dev
#libxml2-dev libflac-dev libmpg123-dev libpng-dev libmpeg2-4-dev libsdl-dev libguichan-dev libsdl-ttf2.0-dev libsdl-gfx1.2-dev \
#libsdl2-dev libsdl2-image-dev libsdl-image1.2-dev

cd "$HOME"
git clone https://github.com/midwan/amiberry -b dev amiberry-dev
cd amiberry-dev

make clean
G_RUN_CMD make -j $(nproc --all) all PLATFORM=rpi3-sdl2

#Run pre-req binaries
G_AGI libmpg123-0 libxml2 mpeg2dec libmpeg2-4

#Xserver/alsa
G_AGI alsa-utils xcompmgr xterm xinit xauth xserver-xorg dbus-x11 xfonts-base x11-xserver-utils x11-common x11-utils

#Set RPi GPU mem 128MB, reboot
#Run/test binary

#AmiBerry compile  -----------------------------------------------------------------
```

**Comment by midwan** [comment]
@Fourdee 
Very important question: What video driver do you have enabled in `/boot/config.txt` ?
KMSDRM will only work with **fkms** or **KMS**.

There's no need to run it under `xinit `from my tests here, but there are a few important details:

Legacy driver: **Dispmanx** can be used, but there's not OpenGL(ES) support (not that we use it in Amiberry)
FKMS driver: **Dispmanx** can be used, and there's also OpenGL(ES) support (like a hybrid).
KMS driver: **Dispmanx** is disabled and **cannot** be used, there's only OpenGL(ES) support.

Considering the above, I have multiple targets available in the Makefile to cover all possible combinations:

If you have Legacy driver: Use the **SDL1 + DispmanX** target
If you have the FKMS driver (Recommended): You can use both **SDL1 + Dispmanx** or **SDL2 + Dispmanx** versions. You can also use the pure SDL2 version, but it is currently slower than Dispmanx on the Pi, so it's not recommended.
If you have the full KMS driver and/or running under x11: Use the **pure SDL2** target (it's the only one that works)

Of course, how you configure SDL2 is important also:

- Use `--enable-video-kmsdrm` and `--disable-video-rpi` only if you're using FKMS or KMS driver. Otherwise it doesn't work anyway. It's important to disable the "rpi" video backend, otherwise it doesn't seem to work with Dispmanx (since we open it directly from Amiberry).

- You don't have to disable x11 or OpenGL if you're using the `fkms` driver, since the `kmsdrm` backend will take higher priority anyway and it can open a screen from the console.

I will update my Wiki to hopefully make this more clear also, but please let me know if the above makes sense and if you run into problems?

**Comment by midwan** [comment]
@Fourdee 
Please note that the `dev` branch is now merged into `master` on my repo. :)
I've also made binaries available for multiple configurations (`sdl1` and `sdl2`). I've compiled the `sdl1` versions on DietPi and they seem to work fine.

Please let me know if you run into any problems! :)

**Comment by Fourdee** [comment]
@midwan 

Was using ```vc4-kms-v3d```, for SDL2. Worked fine with the older AmiBerry binaries, but not the dev compiled one.

Thanks for the info 👍 , i'll recompile and retest from master.

**Comment by midwan** [comment]
@Fourdee 
You can start using the master branch from now on, since the latest stuff is merged there. :)
If you want binaries, those are also available: https://github.com/midwan/amiberry/releases

**Comment by Fourdee** [comment]
@midwan 

Many thanks 👍 

Getting somewhere, user error my end, was using ```dtoverlay=vc4-kms-v3d```, instead of:
```
dtoverlay=vc4-fkms-v3d
```

**Comment by midwan** [comment]
@Fourdee 
Great, let me know if you get stuck anywhere and I can help!

**Comment by Fourdee** [comment]
@midwan 

```libgles2-mesa```

Yay, on screen AmiBerry 😃 . I should be good to go now, just need to update install in DietPi, will let you know if I get stuck 👍 

Your AmiBerry binaries worked a treat :)

**Comment by Fourdee** [comment]
RPi 3: Post test install notes:
- APT firmware/kernel 4.9.59
- KMS (without X11) Using our previous lower FB resolution (optimization), results in 3/4 of screen mouse reach and cursor offset to actual result. FB 1080p fine. ~Could be config related (fullscreen unchecked), see below.~
- Elite using 1/4 of horz space. Probably due to using existing (old) autostart.uae, will create a new default.

🈺 320x200 1/4 horz
🈯️ 640x240 fine

---

Restart of emulator (via restart button), results in 10%~ of previous FPS. Could be driver related, will try with latest vc4 kernel/firmware etc.
- 🈺  Config, CPU speed always resets to 7MHz from Fastest on reload

**Comment by Fourdee** [comment]
@midwan 

Any chance you have a stock/raw ```autostart.uae``` for the latest version of AmiBerry I can test/distribute for our install? I tried clearing our previous config (to only leave in the paths), however, few issues with CPU speed being reset each restart:
https://github.com/Fourdee/DietPi/issues/1410#issuecomment-363589044

**Comment by midwan** [comment]
@Fourdee 
That's why we have "Quickstart" options now, in the GUI itself... :)
If there is no bundled config, the Quickstart options should be enough to get you started with a pre-defined machine configuration, the default being the most common (A500 with 512Kb expansion RAM).

Also, remember to delete any old `adfdir.conf` file if you had it, it will be regenerated by the emulator anyway. We added more stuff there as well.

**Comment by Fourdee** [comment]
Initial 1st pass installation for RPi 2/3 is completed.

Outstanding:
- General testing, identify/resolve any DietPi specific config issues. If not, i'll create Git Ticket on AmiBerry.
- Implement for other devices (i've got x86_64 PC on my personal to-do list @midwan  ;) )

**Comment by midwan** [comment]
Great news!

I'm currently preparing an Android build, following that I'll start on a x86 target as well :)

**Comment by Fourdee** [comment]
@midwan 

> I'm currently preparing an Android build

💘. AmiBerry on the go? Count me in 👍 
I can test on my https://www.gsmarena.com/xiaomi_redmi_note_4x-8580.php (-pink), just let me know.

**Comment by midwan** [comment]
@Fourdee 
Thanks! I'll let you know once I've got something running that's not too embarrassing to share. :-)

**Comment by midwan** [comment]
@Fourdee 
Actually, you can already have Amiberry on the go even without Android:
![27657388_10156230949096579_8878961804717874123_n](https://user-images.githubusercontent.com/3877365/35947764-c0141b24-0c69-11e8-9e52-5738a4af3eb1.jpg)

![27540421_10156230949256579_5056245945475286701_n](https://user-images.githubusercontent.com/3877365/35947768-c5546076-0c69-11e8-9c57-ef1bef5f9c70.jpg)

That's running a Raspberry Pi CM3 module ;)

**Comment by Fourdee** [comment]
@midwan 

> Actually, you can already have Amiberry on the go even without Android:
That's running a Raspberry Pi CM3 module ;)

Simply epic!

**Comment by midwan** [comment]
@Fourdee 
Heads up, just pushed out v2.13 out of beta. :)
The master branch is updated, and binaries are available for all supported distros (Jessie and Stretch) here: https://github.com/midwan/amiberry/releases

**Comment by Fourdee** [comment]
@midwan 
Legend 👍 Many thanks.

RPi1-sdl2 aswell, nice one 😄 

I'll pick this back up next few days. v6.2 release planned before 18th

**Comment by midwan** [comment]
@Fourdee 
Great!
Just a question: Will users of older versions of DietPi (e.g. <6.x) be able to upgrade their Amiberry installations through `dietpi-upgrade`, or is that not supported and they'll have to re-flash a v6.x image first?

**Comment by Fourdee** [comment]
@midwan 

> Just a question: Will users of older versions of DietPi (e.g. <6.x) be able to upgrade their Amiberry installations through dietpi-upgrade, or is that not supported and they'll have to re-flash a v6.x image first?

Bit tricky, due to v159 requiring new image, and, AmiBerry image is Jessie, new one will be Stretch (as per v6.x)
If they are running DietPi v159 (actually v5.59) or lower, they will need to fresh install via latest image regardless. The existing code is not compatible with DietPi in v6.x. 

The other issue is the previous v6.0 AmiBerry DietPi image is Jessie, the new one will be Stretch only for RPi. We now only support 1 image/distro per device (RPi being Stretch as of v6.0). The v6.0 Jessie AmiBerry image was offered as a way for users to install AmiBerry then, whilst waiting for the full RPi Stretch AmiBerry install.

Due to the above, I believe patching is not viable. Its looking like fresh installs in this case, at the moment.

---

Regardless, once the new image is released with latest AmiBerry under Stretch and v6.x. We can patch those installations in the future with updates without issue.

Hope this all makes sense, if not please let me know (rough day).

**Comment by Fourdee** [comment]
Verified SDL2 installs and AmiBerry displaying on screen:
- 🈯️ RPi 3
- 🈯️ RPi Zero

ToDo:
- General usage tests
- 🈯️ Update online docs
- Create new Stretch Amiberry image
- Implement for non-rpi devices (will be done for v6.3, won't make it for v6.2 release this week)

---

Issues:
- ```autostart.uae``` not loaded automatically, -f path? Have to manually load after each run
- I've put autostart.uae into ```/var/lib/dietpi/dietpi-software/installed/amiberry/conf/``` and launch command is 
```
WorkingDirectory=/var/lib/dietpi/dietpi-software/installed/amiberry
ExecStart=/bin/bash -c './$amiberry_filename -f conf/autostart.uae'
```
Still have to manually load each run.

**Comment by midwan** [comment]
@Fourdee 
I'm pushing a new version into master today, v2.14. :)

Binaries will also be released with it, as usual. This time, I'll be creating the binaries using profiler-assisted optimizations, so the performance on those will be as good as possible. This also means that there will be a difference in performance between the binaries (best performance) and compiling from source directly, unless you also use the profiler of course.

To use the profiler, you need to compile the binary twice:
- First, enable the `GEN_PROFILE=1` commented out line in the Makefile, compile as normal, then run the generated binary and use it in a scenario you want to optimize for. I render a benchmark scene in Lightwave for example.
- Then you'll need to do a `make clean`, change the Makefile again to disable `GEN_PROFILE=1` and enable `USE_PROFILE=1` instead. Recompile for the same target and you'll get an optimized binary.
- To clean out the profiler generated files, you can do a `make cleanprofile`.
- Repeat the above process for each target you want to create a binary for.

**Comment by Fourdee** [comment]
@midwan 

> I'm pushing a new version into master today, v2.14. :)
Binaries will also be released with it, as usual. This time, I'll be creating the binaries using profiler-assisted optimizations, so the performance on those will be as good as possible.

Excellent 👍 

I'am just wrapping up v6.2 for release today. I'll make sure to include (and test) the 2.14 binaries before we do.

**Comment by Fourdee** [comment]
@midwan 

2.14 👍 

> Starting with v2.14, these also include the capsimg.so library, to enable IPF support.

~Quick question, I assume ```capsimg.so``` is specific to the binary and/or CPU architecture its bundled with?~

I see the filesizes are different, i'll bundle required  ```capsimg.so```  per device.

NB:
- RPi1 still using 2.13 binary until I get time to recompile, or @midwan can provide it in release section

**Comment by Fourdee** [comment]
@midwan

Some notes with testing, could be an issue DietPi end or not. Unsure at the moment.
 
#### Minor Issue 1:
RPi 3, SDL2:
- Run elite
- F12 
- Select restart
- Reload autostart.uae with previous setting and elite
- selecting "start" hangs with black screen for 10-20 seconds, before starting. Same with 4.9 and 4.14 kernels.

#### Minor issue 2:
- autostart.uae is not loaded upon 1st run or restart of AmiBerry. Manually selecting "load" is required afterwards.

#### Minor issue 3:
- Lack the RPi1 2.14 binary. We'll use the 2.13 for now. I won't get the time to recompile prior to release of DietPi today.

Apart from the above, Elite runs flawless and at a higher FPS than the previous AmiBerry version, great work 👍 🎖 

---

@midwan 
I'll update you with more details on the installation and the RPi AmiBerry image, once DietPi v6.2 is released.

**Comment by midwan** [comment]
@Fourdee 
The RPI1 build has some issues with JIT (segfaults), so I didn't provide a binary until I find a solution.

What does "autostart.uae" contain? Was it generated from Quickstart? If not, I'd advise that it's generated from scratch, using a Quickstart config as a template, since some config values have been changed from older versions of the emulator and perhaps something in there is incorrect...

**Comment by Fourdee** [comment]
@midwan 

> What does "autostart.uae" contain? Was it generated from Quickstart?

I'll recheck/retest

**Comment by Fourdee** [comment]
@midwan 

Image done for RPi, auto installation of latest AmiBerry:
- New DL link (no changes needed to amiberry.com info, only the link change) http://dietpi.com/downloads/images/DietPi_RPi-ARMv6-Stretch_AmiBerry.7z
- Brings image up to Debian Stretch
- SDL2 is the default and only installation method (I've opted to remove SDL1, as SDL2 seems stable and offers improved performance, newer library, less confusion with multiple options for our users etc). If there is any major reason to include SDL1 versions again, please let me know and we can review
- RPi2/3 = 2.1.4, RPi1 = 2.1.3. But we can now patch during DietPi updates with ease.
- 🈯️ RPi 3 test install
- 🈴 Outstanding issue with ```autostart.uae``` not loading automatically. I'll review/retest when I can.
- Other (non-rpi) devices are still pending and will take much longer to implement. I need to do some research 😉 

Any problems, please let me know.

Great work again on 2.1.4, amazing achievement. When I get 5 minutes, i'll be having a good play of Elite 👍 😃

**Comment by midwan** [comment]
@Fourdee 
Thanks, I've updated the link to the download webpage. :)

**Comment by midwan** [comment]
@Fourdee 
Amiberry v2.18 is now online! :)
https://github.com/midwan/amiberry/releases

- Improved audio latency
- Unique new feature: WHDLoad booter! (check the wiki for details)
- Bugfixes in various places

**Comment by Fourdee** [comment]
@midwan 

> Improved audio latency

👍 💘 

Great work as always and thanks for the heads up. We'll patch the 2.18 update for existing installations, in our v6.5 release (ETA 1 week).

**Comment by MichaIng** [comment]
v2.19 up: https://github.com/midwan/amiberry/releases

**Comment by Fourdee** [comment]
> v2.19 up: https://github.com/midwan/amiberry/releases

Was added in v6.8 https://github.com/Fourdee/DietPi/issues/1707#issue-314377609

**Comment by midwan** [comment]
@Fourdee 
Regarding the Tinkerboard, we made some good progress recently and we have a minimal image (based on Armbian) with GPU support that runs very quickly. There's a wiki on how to set it up from scratch here: https://github.com/midwan/amiberry/wiki/Compiling-for-the-Tinker-Board

I think you should be able to use it to get GPU support for DietPi as well, and then the Amiberry version for the Tinkerboard can be included in the list of platforms also :)

**Comment by Fourdee** [comment]
@midwan 

Great news, good to hear 👍 

I'll take a look when I can.

**Comment by Fourdee** [comment]
Will mark this as closed and create new tickets as needed.
